packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -113,44 +113,64 @@ export default class MuiDetailGenerator implements DetailGenerator {
               if (ts.isJsxAttribute(prop)) {
                 const propName = prop.name.escapedText.toString();
                 const inputProp = widgetProperties.properties.find(l => l.name === propName);
-
                 if (inputProp) {
                   if (prop.initializer) {
-                    if (ts.isStringLiteral(prop.initializer)) {
-                      if(inputProp.value !== prop.initializer.text) {
+                    const initializer = prop.initializer
+                    if (inputProp.type === "STRING_LITERAL") {
+                      const text = ts.isStringLiteral(initializer) ? initializer.text : initializer.getText()
+                      if(inputProp.value !== text) {
                         newProp = factory.updateJsxAttribute(prop, prop.name, factory.createStringLiteral(inputProp.value));
                         astChanged = true;
                       }
-                    } else if (ts.isJsxExpression(prop.initializer)) {
-                      if (prop.initializer.expression) {
-                        if (prop.initializer.expression.kind === SyntaxKind.TrueKeyword || prop.initializer.expression.kind === SyntaxKind.FalseKeyword) {
-                          if(inputProp.value !== prop.initializer.expression.getText()) {
-                            const booleanValue = factory.createJsxExpression(
-                              undefined,
-                              inputProp.value.toLowerCase() === 'true' ? factory.createTrue() : factory.createFalse()
-                            )
+                    } else if (inputProp.type === "EXPRESSION") {
+                      const text = ts.isJsxExpression(initializer) ? initializer.expression?.getText() : initializer.text 
+                      if (inputProp.value !== text || !ts.isJsxExpression(initializer)) {
+                        const newAst = createAst(inputProp.value)
+                        const statement = newAst?.statements[0] as any
+                        if (statement) {
+                          const expression = statement.expression
+                          if (expression) {
```

Comment:
just one if - use optional chaining like this: `expression?.kind === SyntaxKind.NumericLiteral`